File: remove_console_logs.py
import os
import re

def remove_console_logs(directory):
    count = 0
    pattern = re.compile(r'^\s*console\.log\s*\(.*?\)\s*;?\s*$', re.MULTILINE)
    
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.next' in dirs:
            dirs.remove('.next')
        if 'build' in dirs:
            dirs.remove('build')
        
        for file in files:
            if file.endswith(('.js', '.jsx', '.ts', '.tsx')):
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content, num_subs = pattern.subn('', content)
                    
                    if num_subs > 0:
                        count += num_subs
                        with open(filepath, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                except Exception as e:
                    pass
    return count

if __name__ == "__main__":
    web_portal_count = remove_console_logs('c:/Projects/dp/web_portal')
    backend_count = remove_console_logs('c:/Projects/dp/backend')
    total = web_portal_count + backend_count
    print(f"Removed {total} console.log statements.")

File: remove_console_logs.js
const fs = require('fs');
const path = require('path');

function removeConsoleLogs(directory) {
    let count = 0;
    const pattern = /^\s*console\.log\s*\(.*?\)\s*;?\s*$/gm;

    if (!fs.existsSync(directory)) return 0;

    const files = fs.readdirSync(directory);
    
    for (const file of files) {
        if (file === 'node_modules' || file === '.next' || file === 'build') continue;
        
        const filepath = path.join(directory, file);
        const stat = fs.statSync(filepath);
        
        if (stat.isDirectory()) {
            count += removeConsoleLogs(filepath);
        } else if (file.endsWith('.js') || file.endsWith('.jsx') || file.endsWith('.ts') || file.endsWith('.tsx')) {
            try {
                let content = fs.readFileSync(filepath, 'utf-8');
                const matches = content.match(pattern);
                if (matches && matches.length > 0) {
                    count += matches.length;
                    content = content.replace(pattern, '');
                    fs.writeFileSync(filepath, content, 'utf-8');
                }
            } catch (e) {}
        }
    }
    return count;
}

const webCount = removeConsoleLogs(path.join(__dirname, 'web_portal'));
const backendCount = removeConsoleLogs(path.join(__dirname, 'backend'));
console.info(`Removed ${webCount + backendCount} console.log statements.`);
